Sources/DarkModeCore/DMTraitCollection.m

```diff
@@ -31,4 +179,69 @@ - (instancetype)init {
   return self;
 }
 
+// MARK: - Observer Registration
++ (void)registerWithApplication:(UIApplication *)application syncImmediately:(BOOL)syncImmediately animated:(BOOL)animated {
+  __weak UIApplication *weakApp = application;
+  _userInterfaceStyleChangeHandler = ^(DMTraitCollection *traitCollection, BOOL animated) {
+    __strong UIApplication *strongApp = weakApp;
+    if (!strongApp)
+      return;
+
+    [self updateUIWithViews:strongApp.windows viewControllers:nil traitCollection:traitCollection animated:animated];
```

Comment:
One thing I overlooked previously. The use of `self` here creates a retain cycle. Even though this retain cycle doesn't matter because it is in a class method, it still is a best practice to avoid retain cycles whenever possible. Maybe use a weak reference to `self` here and in the method below in next PR.